.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,105 @@
+name: Run Unit Tests
+on:
+  pull_request: # Run in every PR that touches .py files
+    paths:
+      - '**.py*'
+  workflow_dispatch: # Allow manually triggering the workflow
+  schedule:
+    # Run roughly every 15 days at 00:00 UTC
+    # (useful to check if updates on dependencies break the package)
+    - cron: "0 0 1,16 * *"
+
+concurrency:
+  group: >-
+    ${{ github.workflow }}-${{ github.ref_type }}-
+    ${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+env: # default versions
+  POETRY_VERSION: 1.3.1
+  PYTHON_VERSION: 3.11.2
+
+permissions:
+  contents: read
+
+jobs:
+  prepare: # step needed to prepare distribution of package used by the test step
+    runs-on: ubuntu-latest
+    outputs:
+      wheel-distribution: ${{ steps.wheel-distribution.outputs.path }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set Python Version
+        run: echo "PYTHON_VERSION=$(cat .python-version)" >> $GITHUB_ENV
+        shell: bash
+    
+      - name: Setup Python and Poetry
+        uses: ./.github/actions/setup-python-poetry
+        with:
+            python-version: ${{env.PYTHON_VERSION}}
+            poetry-version: ${{env.POETRY_VERSION}}
+            install-deps: true
+            only-dev-deps: true
+
+      - name: Build package distribution files
+        run: |
+          poetry build
```

Comment:
nit: compact commands like `poetry build` or `make test` (below) can be on the same line as `run`?